### DISCUSSION TITLE
How do you deal with same host, multiple IP's?

### DISCUSSION THREAD

**Discussion by kylothian** [selivan/https-ssl-cert-check-zabbix#27] [Q&A]
So my domain is hosted behind Cloudflare and IP's rotate. If you navigate in Zabbix to Configuration > Host > Select your host > Items > ssl_cert_check_valid > Test

{$SSLDOMAIN} is populated with my domain name and {$IPADDR} is populated by the first address it gets. Get Value and Test returns a value of 0

If i run my domain via the terminal bash /usr/lib/zabbix/externalscripts/ssl_cert_check.sh valid mydomain.com it returns value of 1, so It can do a lookup using the domain name, but the {$IPADDR} is preventing me from getting the correct data in zabbix console.

If I set  {$IPADDR} the same as {$SSLDOMAIN} value 0 is returned as well, which is false because the domain ssl is valid.

Am I missing something obvious? If I try and remove {$IPADDR} entirely from the Item I am met with the -65535 value

edit : Troubleshooting further, found the following:
(returns correct value)
bash /usr/lib/zabbix/externalscripts/ssl_cert_check.sh valid sha256.badssl.com
1

Going via Site getting the following see attachment: This seems to fail
<img width="829" alt="Screenshot 2021-07-01 at 15 06 53" src="https://user-images.githubusercontent.com/32124039/124129435-1354ef80-da7e-11eb-93cb-a083ee85653d.png">

Any advice would be greatly appreciated!

**Comment by selivan** [comment]
Templates in repository are just examples. If you have different setup, you should create your own template.

If I understand correctly, that you want to test some domain name with every of multiple IP it resolves into, you should write auto-discovery script that returns something like

```
{
{ '{#DOMAIN}': 'domain1', '{#IP}': '1.1.1.1', '{#SSLPORT}': 443 },
{ '{#DOMAIN}': 'domain1', '{#IP}': '1.1.1.2', '{#SSLPORT}': 443 },
{ '{#DOMAIN}': 'domain1', '{#IP}': '1.1.1.3', '{#SSLPORT}': 443 },
{ '{#DOMAIN}': 'domain2', '{#IP}': '2.1.1.1', '{#SSLPORT}': 443 },
{ '{#DOMAIN}': 'domain2', '{#IP}': '2.1.1.2', '{#SSLPORT}': 443 },
}

```

and template with auto-discovery rule using it.

---

If you suspect the script is not working correctly on some host, you should understand what exactly parameters script gets in your setup, so that we can reproduce and debug it. I don't know much how the example templates work, they were submitted by other contributors, so I don't know how to reproduce what the script is getting in your case from the screenshot.
